Tests/SourceKitLSPTests/BuildSystemTests.swift

```diff
@@ -87,35 +90,44 @@ final class BuildSystemTests: XCTestCase {
   var haveClangd: Bool = false
 
   override func setUp() {
-    haveClangd = ToolchainRegistry.shared.toolchains.contains { $0.clangd != nil }
-    testServer = TestSourceKitServer()
-    buildSystem = TestBuildSystem()
-
-    let server = testServer.server!
-    documentManager = server._documentManager
-
-    self.workspace = Workspace(
-      documentManager: DocumentManager(),
-      rootUri: nil,
-      capabilityRegistry: CapabilityRegistry(clientCapabilities: ClientCapabilities()),
-      toolchainRegistry: ToolchainRegistry.shared,
-      buildSetup: TestSourceKitServer.serverOptions.buildSetup,
-      underlyingBuildSystem: buildSystem,
-      index: nil,
-      indexDelegate: nil)
-
-    server._workspaces = [workspace]
-    workspace.buildSystemManager.delegate = server
-
-    sk = testServer.client
-    _ = try! sk.sendSync(InitializeRequest(
+    // XCTestCase.setUp cannot be async, so unfortunately we need to do some
+    // hackery to synchronously wait for a task to finish. This is very much an
+    // anti-pattern because it can easily lead to priority inversions and should
+    // thus not be copied to any non-test code.
+    let sema = DispatchSemaphore(value: 0)
```

Comment:
Could we use an expectation here instead?

Reply:
Good idea. I’ll do that in a follow-up PR to get this one merged.